plugins/woocommerce/includes/data-stores/class-wc-customer-data-store-session.php

```diff
@@ -81,19 +84,41 @@ public function save_to_session( $customer ) {
 			if ( 'billing_' === substr( $session_key, 0, 8 ) ) {
 				$session_key = str_replace( 'billing_', '', $session_key );
 			}
-			$data[ $session_key ] = (string) $customer->{"get_$function_key"}( 'edit' );
+			if ( 'meta_data' === $session_key ) {
```

Comment:
This handles the saving of meta data to session. To avoid saving all meta to session, we have an allowlist of properties. Other meta is discarded like before.